coderd/templateversions.go

```diff
@@ -163,6 +164,215 @@ func (api *API) templateVersionParameters(rw http.ResponseWriter, r *http.Reques
 	httpapi.Write(rw, http.StatusOK, values)
 }
 
+func (api *API) createTemplateVersionPlan(rw http.ResponseWriter, r *http.Request) {
+	apiKey := httpmw.APIKey(r)
+	templateVersion := httpmw.TemplateVersionParam(r)
+	if !api.Authorize(rw, r, rbac.ActionRead, templateVersion) {
+		return
+	}
+	// We use the workspace RBAC check since we don't want to allow plans if the
+	// user can't create workspaces.
+	if !api.Authorize(rw, r, rbac.ActionCreate,
+		rbac.ResourceWorkspace.InOrg(templateVersion.OrganizationID).WithOwner(apiKey.UserID.String())) {
+		return
+	}
+
+	var req codersdk.CreateTemplateVersionPlanRequest
+	if !httpapi.Read(rw, r, &req) {
+		return
+	}
```

Comment:
As a user, I could spam this ~1000 times and take down a Coder deployment for the day. The same could be said for workspaces, but we have issues tracking quotas for those.

Reply:
OK so I'll just add details to those issues that whatever fix we do needs to apply to this endpoint too

Reply:
Could you link the issue? I can't seem to find it after searching "rate limit" and "quota"

Reply:
Multiple builds cannot occur for the same workspace in parallel because they are stateful.

Reply:
Do we have any protection across multiple builds across multiple workspaces on the same user?

Reply:
That's fair, but I still think that a global solution should be applied to all jobs simultaneously. You said earlier there was a ticket about this, could you link it so I can add context to it? I will try to tackle it before community MVP